saltstack/base/salt/dhcp/etc/dhcp/dhclient-enter-hooks

```diff
@@ -69,12 +70,15 @@ make_resolv_conf() {
             for nameserver in ${new_domain_name_servers} ; do
                 REVERSE_DNS=$(dig @${nameserver} -x $IP)
                 log "Reverse DNS by ${nameserver}: ${REVERSE_DNS}"
-                echo "$REVERSE_DNS" | grep -qc "${search}"
+                echo "$REVERSE_DNS" | grep -A 1 "ANSWER SECTION" | grep -qc "${search}"
                 if [[ $? -eq 0 ]]; then
                     set_hostname_from_dns "$REVERSE_DNS" "$IP" "$search"
                     break
                 else
                     log "$nameserver DNS server didn't return valid result for $IP"
+                    # restarting network will force the dhcp client to run again and call make_resolv_conf
+                    /etc/init.d/network restart > $NETWORK_RESTART_LOGFILE 2>&1
```

Comment:
If we don't restart the network daemon what happens?

Reply:
I tried it on RHEL6: unfortunately nothing. Automatic run occurred 25 minutes later ...

Reply:
exit 1 did not have an effect of restarting the script, either.